File: Java/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
import java.lang.Math;
class Solution {
    public int solution(int number, int limit, int power) {
        int sum = 0;
        for(int i=1; i<=number;i++){
            System.out.println(divisor(i));
            sum += divisor(i) > limit ? power : divisor(i);
        }
        return sum;
    }
    int divisor(int n){
     
        int sum = 0;
        int sq = (int)Math.sqrt(n);
        for(int i=1; i<=sq; i++){
            if(n%i==0){
                sum+=1;
                if(n/i != i){
                    sum+=1;
                }
            }
        }
        return sum;
    }
}

File: Kotlin/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.kt
import kotlin.math.sqrt
class Solution {
    fun solution(number: Int, limit: Int, power: Int): Int {
        var answer: Int = 0
        for(i in 1..number) {
            var cnt = 0
            var iSqrt = sqrt(i.toDouble())
            for(j in 1..iSqrt.toInt()) {
                if(i%j==0) {
                    cnt++
                    
                }
            }
            if(iSqrt-iSqrt.toInt()==0.0) {
                cnt = 2*cnt-1
            } else {cnt = 2*cnt}
            if(cnt > limit) {
                cnt = power
            }
            answer+=cnt
            
        }
        return answer
    }
}